finetune/target_models/masked_language_model.py

```diff
@@ -96,6 +99,111 @@ def __init__(self, *args, **kwargs):
+                                                 # PredictMode.MLM_IDS,
+                                                 # PredictMode.MLM_POSITIONS],
+                                                  hooks=hooks)
+        arr_encoded = self.input_pipeline._array_format(encoded)
+        prediction_information = list(predict)
+
+        return prediction_information
+
+
+    def display_text(self, input_text, k=5, context=None, **kwargs):
+        prediction_info = self._inference([input_text],
+                                          predict_keys=[PredictMode.GENERATE_TEXT,
+                                                        PredictMode.MLM_IDS,
+                                                        PredictMode.MLM_POSITIONS],
+                                          context=context,
+                                          force_build_lm=True,
+                                          **kwargs)
+        #print("raw_preds:", raw_preds)
+        #print(np.asarray(raw_preds).shape)
+        #return self.input_pipeline.text_encoder.decode(
+        #    np.asarray(raw_preds)
+        #)
+
+        #text_to_display = list(self.input_pipeline.text_to_tokens_mask(input_text))[0]['tokens'][:,0]
+        #text_to_display = "".join([self.input_pipeline.text_encoder.decoder[token_id] for token_id in text_to_display])
+        #text_to_display = bytearray([self.input_pipeline.text_encoder.byte_decoder[c] for c in text_to_display]).decode("utf-8", errors=self.input_pipeline.text_encoder.errors)
+        #print('Text encoded with _encode:', self.input_pipeline.text_encoder._encode([input_text]))
+        #text = "".join([self.decoder[token_id] for token_id in token_ids])
+        #text = bytearray([self.byte_decoder[c] for c in text]).decode(
+        #    "utf-8", errors=self.errors
+        #)
+        #print(prediction_info)
+        predicted_tokens = [{'prediction_ids': [self.input_pipeline.text_encoder.decode([i]) for i in pred['GEN_TEXT'][-k:][::-1]],
+                             'original_token_id': self.input_pipeline.text_encoder.decode([pred['mlm_ids']]),
+                             'position': pred['mlm_positions']} for pred in prediction_info]
+        #import ipdb; ipdb.set_trace()
+        mask_positions = [i['position'] for i in predicted_tokens]
+
+        tokens = self.input_pipeline.text_encoder._encode([input_text]).tokens[0]
+        #print("tokens:",tokens)
+
+        mask_number = iter(range(len(predicted_tokens)))
+        text_to_display = "".join([tokens[i] if i+1 not in mask_positions
+                                             else '<' + str(next(mask_number)) + '>'
+                                             for i in range(len(tokens))]) + 2*'\n'
+        #print(text_to_display)
+
+        for i in sorted(predicted_tokens, key=lambda x: x['position']):
+            text_to_display += (f"<{mask_positions.index(i['position'])}>\t|{i['original_token_id']}|\t{i['prediction_ids']}\n")
```

Comment:
I think this may format kind of strangely depending on length of tokens? May want to use some of the string functions like `ljust` or `rjust` here instead.

Reply:
Yeah... good point. Improved it a bit!